### ISSUE TITLE
Schema validation fails for product types

### ISSUE THREAD

**Issue by saleebm** [wp-graphql/wp-graphql-woocommerce#164]
**Describe the bug**
Since updating to v0.3.0, Apollo client does not download the schema from my remote server anymore because it cannot validate the schema. To sum up the error message, it says that the different Product Types, ie VariableProduct do not match to the interfaces expected from the Product type (ie expected ProductToProductCategoryConnection for Product.Category but getting VariableProductToProductCategoryConnection).

**To Reproduce**
1. I thought it was related to my install, but then I grabbed the Docker image here and set it up.
2. Then I ran:
`$ apollo client:download-schema --endpoint http://localhost:8091/graphql`
3. And then I got the following error:
```
✖ Saving schema to schema.json
→ Interface field argument Product.upsell(where:) expects type ProductToProductConnectionWhereArgs but SimpleProduct.upsell(where:) is type SimpleProductToProductConnectionWhereArgs.

Error: Interface field Product.categories expects type ProductToProductCategoryConnection but VariableProduct.categories is type VariableProductToProductCategoryConnection.

Interface field argument Product.categories(where:) expects type ProductToProductCategoryConnectionWhereArgs but VariableProduct.categories(where:) is type VariableProductToProductCategoryConnectionWhereArgs
...
Interface field Product.upsell expects type ProductToProductConnection but SimpleProduct.upsell is type SimpleProductToProductConnection.
Interface field argument Product.upsell(where:) expects type ProductToProductConnectionWhereArgs but SimpleProduct.upsell(where:) is type SimpleProductToProductConnectionWhereArgs.
    at assertValidSchema (~/.config/yarn/global/node_modules/graphql/type/validate.js:71:11)
    at assertValidExecutionArguments (~/.config/yarn/global/node_modules/graphql/execution/execute.js:136:35)
    at executeImpl (~/.config/yarn/global/node_modules/graphql/execution/execute.js:86:3)
    at execute (~/.config/yarn/global/node_modules/graphql/execution/execute.js:64:63)
    at Object.introspectionFromSchema (~/.config/yarn/global/node_modules/graphql/utilities/introspectionFromSchema.js:31:37)
    at Task.task (~/.config/yarn/global/node_modules/apollo/lib/commands/client/download-schema.js:15:78)
```

**Expected behavior**
I would expect to get a schema from my client installation. However, I just download the schema manually and it is fine for me since the queries resolve for all the product types well.

**Screenshots**
...

**Desktop (please complete the following information):**
...

**Smartphone (please complete the following information):**
...

**Additional context**
It seems that it's difficult for one to simply validate a schema by any other means than to just run `apollo client:download-schema`, I can't find any tools to help with this besides a schema linter.
I don't if this makes sense, but also it seems that the interface for the extended Product Types just needs to implement the default Product one. I hope to help as much as I am able to. Thank you guys for all your effort and hard work.

**Comment by kidunot89** [comment]
@saleebm This a bug related to the new `Interface` type implemented in WPGraphQL v0.4.0. Myself and @jasonbahl are working to get it solved. You can keep up with this issue [here](https://github.com/wp-graphql/wp-graphql/issues/991). We're working to get it dealt with a soon as possible. Until then, I would recommend rolling back to WPGraphQL **v0.3.8** and WooGraphQL to **v0.2.2**.

**Comment by saleebm** [comment]
This seems to have been resolved. I just pulled the latest version of wp-graphql and it is good.
